### PR TITLE
DOCS: Update protocol doc for missing data type

### PR DESCRIPTION
**Pull request by dengziming**
*More detailed description of your change*
Add missing records(KAFKA-9629) uint16(KAFKA-12180) docs.

*Summary of testing strategy (including rationale)*
No

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dengziming** [comment]
@mumrah @cmccabe , Hello, the 2 data types (records, uint16) are imported by you, PTAL.

File: clients/src/main/resources/common/message/README.md [hunk 1]
```diff
@@ -85,6 +87,8 @@ There are several primitive field types available.
 
 * "bytes": binary data.
 
+* "records": record set used in fetch api and fetch snapshot api
```
Comment (chia7712) [inline clients/src/main/resources/common/message/README.md]: `records` is used by produce data also.

**Review by chia7712 — COMMENTED** [review]
@dengziming thanks for your patch. please take a look at one small comment.

File: clients/src/main/resources/common/message/README.md (hunk 1 above)
Comment (dengziming) [reply clients/src/main/resources/common/message/README.md]: Thank you, I think it's better to remove usage cases or we need to update docs every time we use it, I reworded this doc, PTAL.

File: clients/src/main/resources/common/message/README.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ There are several primitive field types available.
 
 * "int16": a 16-bit integer.
 
+* "uint16": a 16-bit unsigned integer.
```
Comment (chia7712) [inline clients/src/main/resources/common/message/README.md:78]: Could you add document for "unit32"?

Comment (dengziming) [reply clients/src/main/resources/common/message/README.md:78]: Added uint32 and uuid.

File: clients/src/main/resources/common/message/README.md [hunk 3]
```diff
@@ -75,16 +75,24 @@ There are several primitive field types available.
...
 
 * "float64": is a double-precision floating point number (IEEE 754).
 
 * "string": a UTF-8 string.
 
+* "uuid": a type 4 immutable universally unique identifier.
+
 * "bytes": binary data.
 
+* "records": memory record set or file record set.
```
Comment (chia7712) [inline clients/src/main/resources/common/message/README.md]: the supported records type in KRPC is only `MemoryRecords`, right?

Comment (dengziming) [reply clients/src/main/resources/common/message/README.md]: Only `MemoryRecords` are supported when receiving records, but we can also set `FileRecords` when writing records. foe example `KafkaRaftClient.tryCompleteFetchRequest()` will call `partitionData.setRecordSet(log.read(xxx))`

Comment (chia7712) [reply clients/src/main/resources/common/message/README.md]: There are many sub classes for this type as the java type used by KRPC is `BaseRecords`. We do make protocol data carry different sub class in processing data (before serialization). Hence, it seems to me listing sub classes in this document is a bit weird.

Comment (dengziming) [reply clients/src/main/resources/common/message/README.md]: I think we should avoid listing subclasses since we would have to update docs every time we add a new subclass, how about rewording this to "recordset such as memory recordset"?

Comment (chia7712) [reply clients/src/main/resources/common/message/README.md]: That LGTM :)
